### PR TITLE
Use standard library instead of shell calls

### PR DESCRIPTION
**Pull request by AnyOldName3**
This enables the script to work anywhere with working Python 2 (which
should be nowhere anymore, but that's beside the point), even if they
don't have standard Posix tools available. Now this works on Windows
without MSYS/Cygwin, which are too much work for most A-Level students.

### CODE REVIEW

**Comment by trickeydan** [comment]
Hi Chris,

Thanks for your contribution. The code looks fine and seems to work upon my testing. I'll make an enquiry to see whether we can merge this.

We(@srobo/brain-team) are planning an overhaul of most of the kit software before October this year, with the bulk of the work planned to occur this summer when our student volunteers will have more spare time. 

I don't yet know how the format for competitor code will look after this overhaul, but the main goal of the overhaul is to upgrade to Python 3.6+ on everything that we can. Hopefully other system-specific code (e.g shell calls) can be removed during this too.

If you'd like to be involved in further kit development, feel free to get in touch :)

**Comment by AnyOldName3** [comment]
I'm not likely to contribute anything else. I had these changes locally as it removed a slow step in getting code onto the robot for the team I'm leading, so there was no disadvantage to making them available upstream.

File: make-zip [hunk 1]
```diff
@@ -33,11 +33,12 @@ assert os.path.isdir( BASE_DIR )
 # Temporary directory for constructing everything inside
 tmpd = tempfile.mkdtemp( suffix="-pyenv" )
 
-subprocess.check_call( "cp -r %s/* %s" % (BASE_DIR, tmpd),
-                       shell = True )
+# annoyingly, copytree creates the directory instead of merging in Python 2
```
Comment (PeterJCLaw) [inline make-zip:36]: We should make this script Python 3 only at some point.

**Review by PeterJCLaw — APPROVED** [review]
@trickeydan I'm :+1: on us just merging this. I've not tested it, but it looks good.
As we know that we have teams who are interested in continuing to use the kits while there isn't an IDE available, as well as for our own development ease, this feels like a useful thing to merge.

File: make-zip (hunk 1 above)
Comment (trickeydan) [reply make-zip:36]: -> #2

**Review by trickeydan — APPROVED** [review]

**Comment by AnyOldName3** [comment]
While I'm not seeing this here, the approval comment mentioned not working on non-Unix platforms. I'm guessing it's been edited out because it's not accurate, but I'm still going to explicitly state that this PR *does* work on Windows, and that was part of my motivation for making this change.

**Comment by PeterJCLaw** [comment]
Awesome, thanks for the contribution. (I agree that this ought to now work on windows, thanks for fixing it to do so!)

**Pull request merged**